pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/stats/SubscriptionStatsImpl.java

```diff
@@ -103,6 +103,12 @@
     /** Mark that the subscription state is kept in sync across different regions. */
     public boolean isReplicated;
 
+    /** Whether out of order delivery is allowed on the Key_Shared subscription. */
+    public boolean allowOutOfOrderDelivery;
```

Comment:
This should also be added to the `SubscriptionStats` interface. 

Also, since this is specific to KeyShared subscriptions, it might get confusing when using other subscription types. We could instead introduce a sub-interface/sub-object to show the key-shared policy.

Reply:
@merlimat 
> Also, since this is specific to KeyShared subscriptions, it might get confusing when using other subscription types. We could instead introduce a sub-interface/sub-object to show the key-shared policy.

Does that mean we should add the following two?

- `KeySharedSubscriptionStats` interface that extends the `SubscriptionStats` interface
- `KeySharedSubscriptionStatsImpl` class that extends the `SubscriptionStatsImpl` class and implements the `KeySharedSubscriptionStats` interface

Reply:
I actually added the above sub-interface/sub-class, but then the deserialization of the json response didn't seem to work on the pulsar-client-admin side. So I just added getters to the `SubscriptionStats` interface.